### PR TITLE
Add btrfs CheckCommand definition

### PR DESCRIPTION
**Pull request by BarbUk**
Check a btrfs mountpoint health and disk space allocated

### CODE REVIEW

File: itl/plugins-contrib.d/storage.conf [hunk 1]
```diff
@@ -102,3 +102,35 @@ object CheckCommand "ceph" {
 		}
 	}
 }
+
+object CheckCommand "btrfs" {
+    import "plugin-check-command"
+    [ "sudo", PluginDir + "/check_btrfs" ]
```
Comment (mcktr) [inline itl/plugins-contrib.d/storage.conf]: `command = ` is missing, thereby the CI tests are failing.

```suggestion
    command = [ "sudo", PluginDir + "/check_btrfs" ]
```

Comment (BarbUk) [reply itl/plugins-contrib.d/storage.conf]: Sorry for the pebkac, fixed by ef0c68f

**Comment by BarbUk** [comment]
I did not try with capabilities/setuid, let me check and report back to you.

**Comment by BarbUk** [comment]
I tried to play with setcap, but the plugin use python-btrfs, so I don't think I can safely use capabilities to allow the plugin to work without sudo:
```
sudo -u nagios /usr/lib/nagios-btrfs/plugins/check_btrfs -m /home
Traceback (most recent call last):
  File "/usr/lib/nagios-btrfs/plugins/check_btrfs", line 180, in <module>
    sys.exit(main())
  File "/usr/lib/nagios-btrfs/plugins/check_btrfs", line 160, in main
    usage_state, usage_msg, usage_summary = check_usage(args, fs)
  File "/usr/lib/nagios-btrfs/plugins/check_btrfs", line 92, in check_usage
    dev_total, allocated, used, wasted_hard, wasted_soft = btrfs.utils.fs_usage(fs)
  File "/usr/lib/python3/dist-packages/btrfs/utils.py", line 164, in fs_usage
    devices = list(fs.devices())
  File "/usr/lib/python3/dist-packages/btrfs/ctree.py", line 485, in devices
    for header, data in btrfs.ioctl.search_v2(self.fd, tree, min_key, max_key):
  File "/usr/lib/python3/dist-packages/btrfs/ioctl.py", line 265, in search_v2
    fcntl.ioctl(fd, IOC_TREE_SEARCH_V2, buf)
PermissionError: [Errno 1] Operation not permitted
```

Commit b6e97a9 explain how to configure sudoers to allow plugin usage.

**Comment by dnsmichi** [comment]
Please change `nagios` into `icinga` as this matches all distributions except Debian/Ubuntu (where we will take care in the future). 

In terms of sudo - there are check command definitions such as `running_kernel` which implement a helper to decide upon a defined custom variable whether to use sudo or not.

I wouldn't bother with setuid and variants, that's up to the user who decides to use that, and not sudo.

**Comment by BarbUk** [comment]
Hello,
fe42c09 and 4944800 fixes the requested changes.
Is there something else to do for this PR ?

> I wouldn't bother to do tests with setuid and variants, that's up to the user who decides to use that, and not sudo.

I understood that sudo in the itl  and the documentation are enough. Please let me know if this PR need anything else.

**Comment by dnsmichi** [comment]
I'll think about the handling of `use_sudo`, but this will take a while into the new year.

**Comment by BarbUk** [comment]
> I'll think about the handling of use_sudo, but this will take a while into the new year.

yes, there is no urgency. Thanks for your feedback.

**Comment by dnsmichi** [comment]
Will deal with this in a separate step, thanks.

**Pull request merged**